src/app/dataset-view/dataset.component.ts

```diff
@@ -206,6 +217,32 @@ const language = 'typescript';
         );
     }
 
+    public getTitle(node: any): string {
+        if (node.event.__typename === "MetadataEventAddData") {
+            return `${node.event.queryOutputData.interval.end - node.event.queryOutputData.interval.start} new records added`;
```

Comment:
Should be `addedOutputData`, not `queryOutputData` - this is probably why root dataset history is not showing up.